daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md

```diff
@@ -0,0 +1,73 @@
+
+- [Kubernetes]({{< ref kubernetes >}})
+
+### Usage
+
+```bash
+dapr mtls renew-certificate [flags]
+```
+
+### Flags
+
+| Name           | Environment Variable | Default           | Description                                 |
+| -------------- | -------------------- | ----------------- | ------------------------------------------- |
+| `--help`, `-h` |                      |                   | help for renew-certificate
+| `--kubernetes`, `-k` |                      | `false` | supprted platform|                             |
+| `--valid-until`  |                      | 365 days |  Validity for newly created certificates |
+| `--restart`  |                      | false |  Restarts Dapr control plane services (Sentry service, Operator service and Placement server) |
+| `--timeout`  |                      | 300 sec |  The timeout for the certificate renewal process |
+| `--ca-root-certificate`  |                      |  |  User provided root certificate pem file path|
+| `--issuer-public-certificate`  |                      |  |  User provided issuer certificate pem file path|
+| `--issuer-private-key`  |                      |  |  User provided issue private key file path|
+| `--private-key`  |                      |  |  User provided root.key file which is used to generate root certificate|
+
+### Examples
+
+#### Renew certificates by generating fresh new certificates
+Generates new root and issuer certificates for Dapr kubernetes cluster with a default validity of 365 days.
+```bash
+dapr mtls renew-certificate -k
+```
+Generates new root and issuer certificates for kubernetes cluster with a default validity of 365 days and restart the control plane services.
+```bash
+dapr mtls renew-certificate -k --restart
+```
+Generates new root and issuer certificates for kubernetes cluster with a given validity.
+```bash
+dapr mtls renew-certificate -k --valid-until <no of days>
+```
+Generates new root and issuer certificates for kubernetes cluster with a given validity and restart the control place services.
+```bash
+dapr mtls renew-certificate -k --valid-until <no of days> --restart
+```
+#### Renew certificate by using user provided certificates
+Rotates certificate of your kubernetes cluster with provided ca.pem, issuer.pem and issuer.key file path and restart the control plane services
+```bash
+dapr mtls renew-certificate -k --ca-root-certificate <ca.pem> --issuer-private-key <issuer.key> --issuer-public-certificate <issuer.pem> --restart
+```
+Rotates certificate of your kubernetes cluster with provided ca.pem, issuer.pem and issuer.key file path.
+```bash
+dapr mtls renew-certificate -k --ca-root-certificate <ca.pem> --issuer-private-key <issuer.key> --issuer-public-certificate <issuer.pem>
+```
+#### Renew certificates by generating fresh certificates using provided root private key
+Uses existing private root.key to generate new root and issuer certificates for kubernetes cluster with a given validity for created certs.
```

Comment:
```suggestion
Uses existing private root.key to generate new root and issuer certificates for the Kubernetes cluster with a given validity time for created certs.
```

Reply:
What this topic is lacking is any of the guidance on best practices. So next we need to update the guidance in this document https://docs.dapr.io/operations/security/mtls/ especially in the https://docs.dapr.io/operations/security/mtls/#updating-root-or-issuer-certs section. This section was recently updated before these commands were available. I suggest that you talk with Bernd who was the person responsible for testing this out. 

If effect we want be clear that the user has two primary choices and how putting all the mtls commands together you get to choose
1) Use the self-signed Dapr generated root and issuer certs which is easy, but you will encountered downtime on renewal
2) Use this command with a private root key to update the deploy and renew root and issuer certs to avoid downtime. We should show the complete set of steps and commands to accomplish this. 

Bottom line, please work with me and Bernd to update this topic https://docs.dapr.io/operations/security/mtls/ and in particular this section https://docs.dapr.io/operations/security/mtls/#updating-root-or-issuer-certs. For example lets split this topic into Kubernetes and self hosted.

Reply:
@msfussell  As of now these command's implementation is done only for k8s mode. So in the k8s section of https://docs.dapr.io/operations/security/mtls/#updating-root-or-issuer-certs we can update the best practice along with cli command available for that.
I think we can create a separate issue and PR for updating this along with certificate expiration warning messages.
Thoughts?

Reply:
I think we should do this at the same time since it has to be done for v1.7 release anyway. The article above was updated before this new command existed and now we should ensure that the best guidance is explained. These CLI docs do not provide any guidance.

Reply:
@msfussell Update at the same place. The cli commands to be used for each scenario.